hr_holidays_public/tests/test_holidays_calculation.py

```diff
@@ -191,3 +191,37 @@ def test_number_of_hours_excluding_employee_2(self):
 
         self.assertEqual(leave_request.number_of_days, 2)
         self.assertEqual(leave_request.number_of_hours_display, 16)
+
+    def test_list_work_time_per_day_excluding_employee_2(self):
+        self.holiday_type.request_unit = "hour"
+        leave_request = self.HrLeave.new(
```

Comment:
Why are you creating a leave request if it's not used at the end?

Reply:
The leave request is created in a period where there is public holidays. `list_work_time_per_day` should take in account the public holidays. 
The goal of this unit test is to show that `list_work_time_per_day returns` the correct output (2 working days).

Reply:
As said, you are not using it in the unit test. If you remove it, and only pass directly the start and end dates, you will get the same result. Anyways, that's not the main concern. See my comment on the requested changes comment.